pkgs/development/web/nodejs/gyp-patches.nix

```
@@ -0,0 +1,21 @@
+{ fetchpatch }:
+let
+  name = "darwin-sandbox.patch";
+  url = "https://github.com/nodejs/gyp-next/pull/216.patch";
```

please use the commit url, since it's more stable:
```suggestion
  url = "https://github.com/nodejs/gyp-next/commit/43509c3d7d805baa2fb393b0420297d34fe398de.patch";
```
Besides that, LGTM

---

Copied the patch with `file://` URL.

---

> I don’t think GitHub would keep this commit for a long time if I force push to the Node.js PR branch (e.g. to rebase it), so I’d rather wait for PR to be merged (so we can use PR patch URL), or bundle the patch in Nixpkgs PR.

It normally does. In case of doubt, we can use a link from web.archive.org. It's just too much to carry a 207-line patch that might subject to upstream changes in Nixpkgs.

---

>It normally does.

I’m not so sure about this. I couldn’t find any official docs except for https://github.com/isaacs/github/issues/997, and I think I’ve seen some commits (orphaned/unreachable/overwritten by force-push) from old PRs vanish, but I might be mistaken.

---

Using web.archive.org seems fine though. I’ll change it back to the URL.